sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionInfo.java

```diff
@@ -144,6 +155,11 @@ public ExpressionInfo(
                 this.name + "]. It should be a value in " + validGroups + "; however, " +
                 "got [" + group + "].");
         }
+        if (!language.isEmpty() && !validLanguages.contains(language)) {
```

Comment:
if `language` can be empty, how about we use empty to indicate builtin? strictly builtin is not a language.

Reply:
how about we change this to `functionType`, and use "scala_udf", "java_udf", "python_udf", "built-in", "hive". Because using empty to indicate builtin is a little bit risky assumption.

Reply:
sounds good. Hive is not a language either. function type is more correct.

Reply:
so hive -> `hive_udf`

Reply:
what about: udf, hive and built-in or .. udf (scala), udf (java), udf(python), hive and built-in?
Yeah, `functionType` sounds better.